File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr.length; i++) {
            int check = target - arr[i];
            if (map.containsKey(check)) {
                return new int[] { i, map.get(check) };
            }
            map.put(arr[i], i);
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> map;
        int j;
        for(int i = 0; i < nums.size(); ++i){
            map[nums[i]] = i + 1;
        }
        for(int i = 0; i < nums.size(); ++i){
            j = target - nums[i];
            if(map[j] != 0 && map[j] != i + 1){
                return {i, map[j] - 1};
            }

        }
        return {};
    }
};